### ISSUE TITLE
Authentication configuration in Maven's settings.xml

### ISSUE THREAD

**Issue by NBardelot** [fabric8io/docker-maven-plugin#617]
### TLDR ###

Please make the plugin use an "id" field to match the "server" in the settings.xml instead of the technical "registry" field.

### Long version ###

The best-pratice with Maven is to setup the authentication in settings.xml for every server needed.  The "id" field is then used to reference the authentication data ("username" and "password", which by the way is best encrypted using a settings-security.xml file) from the POM files. The best example of this is the maven-deploy-plugin which uses the "id" field of the "repositoy" under "distributionManagement" (in the POM) to match the right server (in the settings). 

In the case of docker-maven-plugin I have a "server" configured (in the settings) that corresponds to my Nexus 3 server, which hosts my Docker private registry. But there is nowhere in the docker-maven-plugin documentation a configuration of the plugin that mentions an "id", or the use of the "distributionManagement". All I have deduced, is that it uses the registry name to match the "id" of the server. Which in my opinion is bad, as in the one hand you have an infrastructure related data (the registry, which can be an IPv4, an IPv6, a name, and can change). And in the other hand you have a purely logical data (the "id" in settings.xml) which has no sense outside of the projet's configuration and can be any string of characters you want, which is very practical as you will not change it any soon (calling it "private-docker-registry" is quite safe and will stand even in case of IP change, server name change etc.).

**Comment by burtsevyg** [comment]
+1
